app/uk/gov/hmrc/helptosavefrontend/models/HTSEvent.scala

```diff
@@ -75,12 +75,14 @@ case class AccountCreated(userInfo: NSIUserInfo)(implicit hc: HeaderCarrier) ext
   )
 }
 
-case class EligibilityResult(nino: NINO, errorDescription: Option[String])(implicit hc: HeaderCarrier) extends HTSEvent {
+case class EligibilityResult(nino: NINO, reason: String, isEligible: Boolean = true)(implicit hc: HeaderCarrier) extends HTSEvent {
   val value: DataEvent = HTSEvent(
     "EligibilityResult", {
-      val basicMap = Map[String, String]("nino" -> nino, "eligible" -> "true")
-      errorDescription.fold(basicMap) { reason ⇒
-        basicMap + ("eligible" -> "false") + ("reason" -> reason)
+      val details = Map[String, String]("nino" -> nino, "eligible" -> "true")
+      if (isEligible) {
+        details + ("reason" -> reason)
+      } else {
+        details + ("eligible" -> "false", "reason" -> reason)
```

Comment:
`reason` is common to both cases - it could go in `details`

Reply:
all fixed, please have another look